Corona-Warn-App/src/test/java/de/rki/coronawarnapp/coronatest/storage/TestCertificateStorageTest.kt

```diff
@@ -0,0 +1,110 @@
+package de.rki.coronawarnapp.coronatest.storage
+
+import android.content.Context
+import androidx.core.content.edit
+import de.rki.coronawarnapp.coronatest.CoronaTestTestData
+import de.rki.coronawarnapp.coronatest.DaggerCoronaTestTestComponent
+import de.rki.coronawarnapp.util.serialization.SerializationModule
+import de.rki.coronawarnapp.vaccination.core.repository.storage.ContainerPostProcessor
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import testhelpers.BaseTest
+import testhelpers.extensions.toComparableJsonPretty
+import testhelpers.preferences.MockSharedPreferences
+import javax.inject.Inject
+
+@Suppress("MaxLineLength")
+class TestCertificateStorageTest : BaseTest() {
+    @MockK lateinit var context: Context
+    private lateinit var mockPreferences: MockSharedPreferences
+    @Inject lateinit var testData: CoronaTestTestData
+    @Inject lateinit var postProcessor: ContainerPostProcessor
```

Comment:
To test the actual `TestCertificateQRCodeExtractor`.
If we mock it all, then we don't know if it actually works in production.
If we manually construct it without DI, we can't be sure that it's the same `TestCertificateQRCodeExtractor` as it would be injected outside of unit tests.